### PR TITLE
10 of the most annoying UI/UX mistakes to avoid

### LINKED ISSUE #3088: 10 of the most annoying UI/UX mistakes to avoid while crafting your products
### Proposed title of article
10 of the most annoying UI/UX mistakes to avoid while crafting your products

### Introduction paragraph (2-3 paragraphs):
The user interface(UI) plays a major role in how a user interacts with your products. From mobile apps, to web apps, to desktop applications, there are tiny details that make it hard or easy for a user to be served by your product. No matter how good your product logic is, the user only interacts with the UI. Good UI/UX practices allow users to spend more time using your product, it increases the chances of the user recommending the product to their colleagues, while bad practices do the exact opposite.

There are some details which when put into consideration, can improve your products' effectiveness. When neglected, they can be the only thing that made your awesome startup fail.

In this article, we look at 10 of the most annoying UI/UX mistakes to avoid while crafting your products.

### Key takeaways: 
We will touch on:
1. Images and image quality
2. Text and the reading area
3. Actions and notification
4. Proper layouts
5. Color palletes
6. Buttons
7. Clickability and clickable areas
8. Inconsistency in design
9. Fonts
10. Data validation

### References: 
N/A

### PR DESCRIPTION
**Pull request by wanjirualice6194**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**
- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**
- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [ ] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [ ] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [ ] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [ ] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?
This closes #3088

### CODE REVIEW

**Comment by geoffrey45** [comment]
@wanjirualice6194 Welcome to the EngEd program 🎉.

Thank you for submitting your article for review.

File: content/articles/10-ui-ux-mistakes/index.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+You should wait for the user to finish typing in a field before showing the validation errors. The errors should be shown only after the user has clicked outside the field, but not after every keystroke.
+
+### 4. Design inconsistencies
+You should use consistent styling to your components. If you decide to use a border-radius of 20, stick to that all through.
+
+![design inconsistencies](inconsistent-design.png)
+
+Using too much colors in your design is also a bad idea. It makes it hard to keep related components together.
+
+You can get balanced color palletes from <https://coolors.co>
```
Comment (geoffrey45) [inline content/articles/10-ui-ux-mistakes/index.md:39]:
```suggestion
You can get balanced color palletes from [coolors.co](https://coolors.co)
```

File: content/articles/10-ui-ux-mistakes/index.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+If you have a multi-platform app, the appearance of the app on all the platforms should be consistent.
+
+### 5. Very wide paragraph
+Having a very wide paragrah that spans more than 2/3 the width of the screen makes it hard to read.
+
+Large paragraphs make it hard to follow a sentence to the next line. Paragraphs should take a maximum of 2/3 of the width of the screen or less. They should also be centered to help the users concentrate on what they are reading and to be able to follow the content line after line.
```
Comment (geoffrey45) [inline content/articles/10-ui-ux-mistakes/index.md:46]: There are different screen sizes in various devices. The statement "2/3 of the screen" is misleading. Please use another measurement unit to drive your point.

**Review by geoffrey45 — CHANGES_REQUESTED** [review]
@wanjirualice6194 Very nice article.

Please check the comments below.

**Comment by hectorkambow** [comment]
plagiarism check done

**Comment by wanjirualice6194** [comment]
Hello @geoffrey45 I have made the requested changes

**Review by geoffrey45 — APPROVED** [review]

**Comment by geoffrey45** [comment]
@wanjirualice6194 Very nice article. Very clean. Keep up the good work 🔥

**Comment by WanjaMIKE** [comment]
Merged #3408
